src/Network/NWProxyConfig.cs

```diff
@@ -0,0 +1,239 @@
+
+			using var resourcePathPtr = new TransientString (relaysResourcePath);
+			var pinned = GCHandle.Alloc (gatewayKeyConfig, GCHandleType.Pinned);
+
+			unsafe {
+				try {
+					var handle = nw_proxy_config_create_oblivious_http (hop!.GetCheckedHandle (), resourcePathPtr, pinned.AddrOfPinnedObject (), (nuint) gatewayKeyConfig!.Length);
+					if (handle == NativeHandle.Zero)
+						return default;
+					return new NWProxyConfig (handle, owns: true);
+				} finally {
+					pinned.Free ();
+				}
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_proxy_config nw_proxy_config_create_http_connect (OS_nw_endpoint proxy_endpoint, /* [NullAllowed] */ OS_nw_protocol_options proxy_tls_options);
+
+		public static NWProxyConfig? CreateHttpConnect (NWEndpoint proxyEndpoint, NWProtocolTlsOptions? options)
+		{
+			if (proxyEndpoint is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (proxyEndpoint));
+
+			var handle = nw_proxy_config_create_http_connect (proxyEndpoint!.GetCheckedHandle (),
+				options?.GetHandle () ?? IntPtr.Zero);
+			if (handle == NativeHandle.Zero)
+				return default;
+			return new NWProxyConfig (handle, true);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_proxy_config nw_proxy_config_create_socksv5 (OS_nw_endpoint proxy_endpoint);
+
+		public static NWProxyConfig? CreateSocksV5 (NWEndpoint endpoint)
+		{
+			if (endpoint is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (endpoint));
+			var handle = nw_proxy_config_create_socksv5 (endpoint!.GetCheckedHandle ());
+			if (handle == NativeHandle.Zero)
+				return default;
+			return new NWProxyConfig (handle, true);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_proxy_config_set_username_and_password (OS_nw_proxy_config proxy_config, /* sbyte* */ IntPtr username, /* sbyte* */ IntPtr password);
+
+		public void SetUsernameAndPassword (string username, string password)
+		{
+			if (username is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (username));
+			if (password is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (password));
+
+			using var usernamePtr = new TransientString (username);
+			using var passwordPtr = new TransientString (password);
+			nw_proxy_config_set_username_and_password (GetCheckedHandle (), usernamePtr, passwordPtr);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_proxy_config_set_failover_allowed (OS_nw_proxy_config proxy_config, /* bool */ byte failover_allowed);
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern byte nw_proxy_config_get_failover_allowed (OS_nw_proxy_config proxy_config);
+
+		public bool FailoverAllowed {
+			get => nw_proxy_config_get_failover_allowed (GetCheckedHandle ()) != 0;
+			set => nw_proxy_config_set_failover_allowed (GetCheckedHandle (), value ? (byte) 1 : (byte) 0);
```

Comment:
```suggestion
			set => nw_proxy_config_set_failover_allowed (GetCheckedHandle (), (byte) (value ? 1 : 0));
```

😄

Reply:
haha 😁 sorry[ I had to check this](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQBIf+OyAzAARrnoDsOA3jrgyaQEYCewApqQLICGAOwCunCKQAULAPZSxANwCUpALwA+UnNIB+Cey5L0pELo6clABgDcTMnu4AlWQDNdM+UrUn9EzTsPHzBWtsAF8gA) since I would assume they were identical

```
public class Program
{
	public byte Manuel (bool v) => v ? (byte) 1 : (byte) 0;
	public byte Rolf (bool v) => (byte) (v ? 1 : 0);
}
```

```msil
// Methods
    .method public hidebysig 
        instance uint8 Manuel (
            bool v
        ) cil managed 
    {
        // Method begins at RVA 0x2069
        // Code size 8 (0x8)
        .maxstack 8

        IL_0000: ldarg.1
        IL_0001: brtrue.s IL_0006

        IL_0003: ldc.i4.0
        IL_0004: br.s IL_0007

        IL_0006: ldc.i4.1

        IL_0007: ret
    } // end of method Program::Manuel
```

```msil
    .method public hidebysig 
        instance uint8 Rolf (
            bool v
        ) cil managed 
    {
        // Method begins at RVA 0x2072
        // Code size 9 (0x9)
        .maxstack 8

        IL_0000: ldarg.1
        IL_0001: brtrue.s IL_0006

        IL_0003: ldc.i4.0
        IL_0004: br.s IL_0007

        IL_0006: ldc.i4.1

        IL_0007: conv.u1
        IL_0008: ret
    } // end of method Program::Rolf
```

There is an extra `IL_0007: conv.u1` on @rolfbjarne's suggestion next would be to check the AOT'ed code 🤣

Reply:
Nah, this is an ongoing between the two of us hahaha

@rolfbjarne touche! https://github.com/xamarin/xamarin-macios/pull/19399#pullrequestreview-1719050599